src/coordinated_workers/coordinator.py

```diff
@@ -807,31 +831,42 @@ def _render_workers_alert_rules(self):
                 "charm_name": worker_topology["charm_name"],
             }
             topology = cosl.JujuTopology.from_dict(topology_dict)
-            alert_rules = cosl.AlertRules(query_type="promql", topology=topology)
-            alert_rules.add_path(WORKER_ORIGINAL_ALERT_RULES_PATH, recursive=True)
-            alert_rules_contents = yaml.dump(alert_rules.as_dict())
 
-            file_name = (
-                f"{CONSOLIDATED_ALERT_RULES_PATH}/rendered_{worker_topology['application']}.rules"
-            )
-            with open(file_name, "w") as writer:
-                writer.write(alert_rules_contents)
+            for orig_path, consolidated_path, type in alert_rules_sources:
+                alert_rules = cosl.AlertRules(query_type=type, topology=topology)  # type: ignore
+                alert_rules.add_path(orig_path)
+                alert_rules_contents = yaml.dump(alert_rules.as_dict())
 
-    def _remove_rendered_alert_rules(self):
-        files = glob.glob(f"{CONSOLIDATED_ALERT_RULES_PATH}/rendered_*")
+                file_name = f"{consolidated_path}/rendered_{worker_topology['application']}.rules"
+                with open(file_name, "w") as writer:
```

Comment:
I think a lot of the new code could be simplified if you used `pathlib.Path` instead of strings + os.path and glob.glob.

E.g. this could be:

```
filename = consolidated_path/f"rendered_{app_name}.rules"
filename.write_text(yaml.safe_dump(alert_rules.as_dict()))
```